internal/scm/github/github.go

```diff
@@ -357,6 +358,31 @@ func (g *Github) GetPullRequests(ctx context.Context, branchName string) ([]scm.
 		return nil, err
 	}
 
+	wg, ctx := errgroup.WithContext(ctx)
+	batches := chunkSlice(repos, 50)
+	var pullRequests []scm.PullRequest
+
+	for _, repos := range batches {
+		repos := repos
+		wg.Go(func() error {
+			result, err := g.getPullRequests(ctx, branchName, repos)
+			if err != nil {
+				return fmt.Errorf("failed to get pull request batch: %w", err)
+			}
+
+			pullRequests = append(pullRequests, result...)
```

Comment:
Not really relevant since we should not do this is concurrent, but writes and reads of this line to `pullRequests` could happen from different go-routines, causing bugs. Running with `-race` should warn here.

A lock would have been one solution, but creating `pullRequests` with a fixed size, and then adding to the exact positions of it from each go-routine is my personally prefered way to solve this to avoid using a lock,